kotlin/download-preplanned-map-area/src/main/java/com/esri/arcgisruntime/sample/downloadpreplannedmaparea/MainActivity.kt

```diff
@@ -0,0 +1,355 @@
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Viewpoint
+import com.esri.arcgisruntime.mapping.view.Graphic
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay
+import com.esri.arcgisruntime.portal.Portal
+import com.esri.arcgisruntime.portal.PortalItem
+import com.esri.arcgisruntime.security.AuthenticationManager
+import com.esri.arcgisruntime.security.DefaultAuthenticationChallengeHandler
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol
+import com.esri.arcgisruntime.symbology.SimpleRenderer
+import com.esri.arcgisruntime.tasks.offlinemap.DownloadPreplannedOfflineMapJob
+import com.esri.arcgisruntime.tasks.offlinemap.OfflineMapTask
+import com.esri.arcgisruntime.tasks.offlinemap.PreplannedMapArea
+import com.esri.arcgisruntime.tasks.offlinemap.PreplannedUpdateMode
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.dialog_layout.*
+import kotlinx.android.synthetic.main.layout_offline_controls.*
+import java.io.File
+import java.util.ArrayList
+
+class MainActivity : AppCompatActivity() {
+
+  private val TAG = MainActivity::class.java.simpleName
+
+  private val offlineMapDirectory by lazy { File(externalCacheDir?.path + getString(R.string.preplanned_offline_map_dir)) }
+  private var preplannedMapAreasAdapter: ArrayAdapter<String>? = null
+  private val downloadedMapAreaNames by lazy { mutableListOf<String>() }
+  private var downloadedMapAreasAdapter: ArrayAdapter<String>? = null
+  private val downloadedMapAreas by lazy { mutableListOf<ArcGISMap>() }
+  private var dialog: Dialog? = null
+
+  private var selectedPreplannedMapArea: PreplannedMapArea? = null
+  private var downloadPreplannedOfflineMapJob: DownloadPreplannedOfflineMapJob? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // delete any previous instances of downloaded maps
+    externalCacheDir?.deleteRecursively()
+
+    // create up a temporary directory in the app's cache when required
+    offlineMapDirectory.also {
+      when {
+        it.mkdirs() -> Log.i(TAG, "Created directory for offline map in " + it.path)
+        it.exists() -> Log.i(TAG, "Offline map directory already exists at " + it.path)
+        else -> Log.e(TAG, "Error creating offline map directory at: " + it.path)
+      }
+    }
+
+    // set the authentication manager to handle challenges when accessing the portal
+    // Note: The sample data is publicly available, so you shouldn't be challenged
+    AuthenticationManager.setAuthenticationChallengeHandler(
+      DefaultAuthenticationChallengeHandler(this)
+    )
+
+    // create a portal to ArcGIS Online
+    val portal = Portal(getString(R.string.arcgis_online_url))
+    // create a portal item using the portal and the item id of a map service
+    val portalItem = PortalItem(portal, getString(R.string.naperville_water_network_url))
```

Comment:
Perhaps its best to rename this string resource to something like `naperville_water_network_id`